core/src/main/java/com/sequenceiq/cloudbreak/service/stack/StackService.java

```diff
@@ -717,6 +717,14 @@ public void validateOrchestrator(Orchestrator orchestrator) {
         }
     }
 
+    public void checkStackExistenceByName(String name, String accountId, Set<String> entries, StackType stackType) {
+        ShowTerminatedClustersAfterConfig showTerminatedClustersAfterConfig = showTerminatedClusterConfigService.get();
```

Comment:
why not just the live clusters?

Reply:
actually it will depend on                     `userProfileService.getOrCreateForLoggedInUser().getShowTerminatedClustersPreferences();` I didn't want to set it to false rather than set it according to preference of the user.

Reply:
The `ShowTerminatedClustersAfterConfig.isActive()` flag is only for internal testing purposes. Even if the terminated stack is found and returned by `findByNameAndWorkspaceId()`, it makes no sense to permit executing operations like `DistroXUpgradeV1Controller.upgradeRdsByName()` for terminated stacks.

In short, I agree with Ricsi that only live clusters should be considered here. (Perhaps the method name could be tweaked accordingly, e.g. `checkLiveStackExistenceByName()`, but this is not essential.)